packages/wouter/src/use-browser-location.js

```diff
@@ -33,6 +38,26 @@ const currentSearch = () => location.search;
 export const useSearch = ({ ssrSearch = "" } = {}) =>
   useLocationProperty(currentSearch, () => ssrSearch);
 
+export const useSearchParams = ({ ssrSearch = "" } = {}) => {
+  const search = useSearch({ ssrSearch });
+  const searchParamsRef = useRef(new URLSearchParams(search));
+  searchParamsRef.current = useMemo(
+    () => new URLSearchParams(search),
+    [search]
+  );
+
+  const setSearchParams = useCallback((nextInit, navOpts) => {
+    const newSearchParams = new URLSearchParams(
+      typeof nextInit === "function"
+        ? nextInit(searchParamsRef.current)
+        : nextInit
+    );
+    navigate("?" + newSearchParams, navOpts);
+  }, []);
+
+  return [searchParamsRef.current, setSearchParams];
+};
+
```

Comment:
Just some ideas to reduce extra React imports and make this function smaller:

```suggestion
export const useSearchParams = ({ ssrSearch = "" } = {}) => {
  const search = useSearch({ ssrSearch });
  const params = useMemo(() => new URLSearchParams(search)));
  
  const setSearchParams = useEvent((nextInit, navOpts) => {
    const newSearchParams = new URLSearchParams(
      typeof nextInit === "function"
        ? nextInit(params)
        : nextInit
    );
    
    navigate("?" + newSearchParams, navOpts);
  });

  return [params, setSearchParams];
};
```

Here, I replaced `useCallback` with `useEvent`. We should not see degraded performance, unless the component that uses it renders too often, e.g. once every 50-300 ms (but that isn't really our responsibility, since when used standalone this hook will re-render only when the query string changes, which in real-life scenarios shouldn't be too often). Plus, `useEvent` is already part of `react-deps.js`.

I'm still trying to figure out how `useMemo` can be avoided further on. The question is whether we really want to memoise it. Considering, that the hook doesn't render too often, it should be safe to always return a new instance of `URLSearchParams`. Why:
1. **Immutability.** Is it safe to re-use this object? What if developer decides to mutate it for some reason, this would cause some undefined bevahiour. 
2. **The memory/CPU footprint of `URLSearchParams`.**  I'm not sure how browsers implement it, but at the first glance it doesn't seem like a heavy resource. Though, I will make a proper research to figure this out.
3. **Stability.** One of the caveats could be that we always return a new object reference. This could harm performance only if `searchParams` instance is further passed down to components as a prop. But in reality, why would someone want to do that? I assume that the normal use-case is the get the object, and then extract the required primitive objects from it, e.g. `searchParams.get("foo")`

Anyways, there is a trade-off that we should evaluate there.